### ISSUE TITLE
Helm 3: issue with registries and nested templates

### ISSUE THREAD

**Issue by jdolitsky** [helm/helm#5757]
To reproduce, first fetch a chart with Helm 2 containing chart depends (wordpress->mariadb):
```
$ helm repo add bitnami https://charts.bitnami.com
"bitnami" has been added to your repositories
$ helm fetch bitnami/wordpress --untar
```

Save the chart with Helm 3, remove original dir
```
$ helm chart save wordpress/ bundle.bar/helm/hub/bitnami/wordpress:5.9.4-alpha1
Name: wordpress
Version: 5.9.4
Meta: sha256:6a8989c24d65acf4d322eaa31ada333a8112d94176bdcac83f8bca50f5fdd430
Content: sha256:6b11d05a224109f4d988247716c6f398b3ff6ca21ebb4ddc9b39243ac511aee9
5.9.4-alpha1: saved
$ rm -rf wordpress/
```

Exporting is met with an error:
```
$ helm chart export bundle.bar/helm/hub/bitnami/wordpress:5.9.4-alpha1
Name: wordpress
Version: 5.9.4
Meta: sha256:6a8989c24d65acf4d322eaa31ada333a8112d94176bdcac83f8bca50f5fdd430
Content: sha256:6b11d05a224109f4d988247716c6f398b3ff6ca21ebb4ddc9b39243ac511aee9
Error: open wordpress/templates/tests/test-mariadb-connection.yaml: no such file or directory
```

**Comment by jdolitsky** [comment]
Note: this issue is actually due to nested directories under `templates/` and nothing to due with dependencies

**Comment by jdolitsky** [comment]
Upon removing the `wordpress/templates/tests/`, everything works as expected and mariadb chart .tgz is present on export

**Comment by bacongobbler** [comment]
@jdolitsky are you planning to PR a fix this week? If you want to ship a fix for the beta, you're more than welcome to, but if you're not planning on PR'ing a fix I'd like to de-prioritize this and move it to the 3.0.0 milestone. Bugfixes shouldn't hold up the beta release; those can be fixed during the beta release cycle. :)

**Comment by jdolitsky** [comment]
@bacongobbler I'm planning to, but we shall see where I get (I'm not sure how involved it might be). If it appears to be too much, I'll switch the milestone asap.

Maybe related to this TODO from 2016? https://github.com/helm/helm/blob/dev-v3/pkg/chartutil/save.go#L205

**Comment by jdolitsky** [comment]
may have figured it out...

**Comment by bacongobbler** [comment]
fixed in #6177
